### PR TITLE
KFSPTS-8215 fix concur authentication filter

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/concur/rest/authentication/ConcurAuthenticationFilter.java [hunk 1]
```diff
@@ -34,13 +34,18 @@ public void doFilter(ServletRequest servletRequest, ServletResponse response, Fi
...
         if (StringUtils.isBlank(authorizationHeader)) {          
             httpServletResponse.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            LOG.debug("doFilter(): Concur authentication failed");
+            return;
         } else {
             final String encodedUserNameAndPasswordAsToken = authorizationHeader.replaceFirst(ConcurConstants.BASIC_AUTHENTICATION_SCHEME + " ", "");
 
             if (!getConcurAuthenticationService().isConcurTokenValid(encodedUserNameAndPasswordAsToken)) {
                 httpServletResponse.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+                LOG.debug("doFilter(): Concur authentication failed");
```
Comment (jhulslander) [inline src/main/java/edu/cornell/kfs/concur/rest/authentication/ConcurAuthenticationFilter.java]: Do we care to know how the login failed (no auth token versus wrong auth token)?  If so, maybe we want different debug statements.  I would be inclined to think they might be errors that we would want in the log.

Comment (db483) [inline src/main/java/edu/cornell/kfs/concur/rest/authentication/ConcurAuthenticationFilter.java]: Good point, I will add more info in the log. Thank you!

**Comment by db483** [comment]
I updated the code to include more information about why authentication failed. This is ready for review again. Thank you.

File: src/main/java/edu/cornell/kfs/concur/rest/authentication/ConcurAuthenticationFilter.java (hunk 1 above)
Comment (db483) [inline src/main/java/edu/cornell/kfs/concur/rest/authentication/ConcurAuthenticationFilter.java]: Let me change them to error too.

**Comment by db483** [comment]
Ready for review.

**Comment by jhulslander** [comment]
Thanks, I will merge after the unit tests run

**Pull request merged**